java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -7576,4 +7580,319 @@ public void geosearchstore(BaseClient client) {
+        String key2 = "{key}-2" + UUID.randomUUID();
+        String initialCursor = "0";
+        long defaultCount = 10;
+        String[] numberMembers = new String[125];
+        for (int i = 0; i < numberMembers.length; i++) {
+            numberMembers[i] = String.valueOf(i);
+        }
+        Set<String> numberMembersSet = Set.of(numberMembers);
+        String[] charMembers = new String[] {"a", "b", "c", "d", "e"};
+        Set<String> charMemberSet = Set.of(charMembers);
+        int resultCursorIndex = 0;
+        int resultCollectionIndex = 1;
+
+        // Empty set
+        Object[] result = client.sscan(key1, initialCursor).get();
+        assertEquals(initialCursor, result[resultCursorIndex]);
+        assertDeepEquals(new String[] {}, result[resultCollectionIndex]);
+
+        // Negative cursor
+        result = client.sscan(key1, "-1").get();
+        assertEquals(initialCursor, result[resultCursorIndex]);
+        assertDeepEquals(new String[] {}, result[resultCollectionIndex]);
+
+        // Negative cursor
+        result = client.sscan(key1, "-1").get();
+        assertEquals(initialCursor, result[resultCursorIndex]);
+        assertDeepEquals(new String[] {}, result[resultCollectionIndex]);
+
+        // Negative cursor
+        result = client.sscan(key1, "-1").get();
+        assertEquals(initialCursor, result[resultCursorIndex]);
+        assertDeepEquals(new String[] {}, result[resultCollectionIndex]);
+
+        // Result contains the whole set
+        assertEquals(charMembers.length, client.sadd(key1, charMembers).get());
+        result = client.sscan(key1, initialCursor).get();
+        assertEquals(initialCursor, result[resultCursorIndex]);
+        assertEquals(charMembers.length, ((Object[]) result[resultCollectionIndex]).length);
+        final Set<Object> resultMembers =
+                Arrays.stream((Object[]) result[resultCollectionIndex]).collect(Collectors.toSet());
+        assertTrue(
+                resultMembers.containsAll(charMemberSet),
+                String.format("resultMembers: {%s}, charMemberSet: {%s}", resultMembers, charMemberSet));
+
+        result =
+                client.sscan(key1, initialCursor, SScanOptions.builder().matchPattern("a").build()).get();
+        assertEquals(initialCursor, result[resultCursorIndex]);
+        assertDeepEquals(new String[] {"a"}, result[resultCollectionIndex]);
+
+        // Result contains a subset of the key
+        assertEquals(numberMembers.length, client.sadd(key1, numberMembers).get());
+        String resultCursor = "0";
+        final Set<Object> secondResultValues = new HashSet<>();
+        do {
+            result = client.sscan(key1, resultCursor).get();
+            resultCursor = result[resultCursorIndex].toString();
+            secondResultValues.addAll(
+                    Arrays.stream((Object[]) result[resultCollectionIndex]).collect(Collectors.toSet()));
+
+            if (resultCursor.equals("0")) {
```

Comment:
fail test instead?  This shouldn't happen.

Reply:
I think this could still happen depending if Redis decides if the set is large and complex enough to split up.